src/dll.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""File implements a simple doubly linked list in python."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Class implements Nodes for use in our dll."""
+    def __init__(self, data=None, nxt=None, prev=None):
+        self.data = data
+        self._next = nxt
+        self._prev = prev
+
+
+class DoublyLinkedList(object):
+    """Class implements a simple doubly linked list."""
+    def __init__(self, iterable=None):
+        self.head = None
+        self.tail = None
+        try:
```

Comment:
I thought we looked at this in person already; it would be much more Pythonic to say `if iterable is not None` before the try block, so you are catching specifically the user's input. With the if statement inside the try block, you are mixing together competing "ask for permission" and "beg forgiveness" approaches.